ieee754r/fmaxmagd32.c

```diff
@@ -0,0 +1,64 @@
+/* The fmaxmag functions determine the value of their argument 
+   whose magnitude is the maximum of  the  magnitudes  of  the  arguments:
+   the value of x if |x| > |y|, y if |x| < |y|, and fmax(x,y) otherwise.
+
+   Copyright (C) 2007-2019 Free Software Foundation, Inc.
+
+   This file is part of the Decimal Floating Point C Library.
+
+   Author(s): Pedro Caldeira <caldeira@linux.ibm.com>
+
+   The Decimal Floating Point C Library is free software; you can
+   redistribute it and/or modify it under the terms of the GNU Lesser
+   General Public License version 2.1.
+
+   The Decimal Floating Point C Library is distributed in the hope that
+   it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+   the GNU Lesser General Public License version 2.1 for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License version 2.1 along with the Decimal Floating Point C Library;
+   if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+   Suite 330, Boston, MA 02111-1307 USA.
+
+   Please see libdfp/COPYING.txt for more information.  */
+
+#ifndef _DECIMAL_SIZE
+#  define _DECIMAL_SIZE 32
+#  include <decimal32.h>
+#endif
+
+#include <math.h>
+#include <ieee754r_private.h>
+
+#define FUNCTION_NAME fmaxmag
+
+#include <dfpmacro.h>
+#include "convert_helpers.h"
+#include <stdio.h>
+#include <limits.h>
+
+DEC_TYPE
+INTERNAL_FUNCTION_NAME (DEC_TYPE x, DEC_TYPE y)
+{
+
+  if( FUNC_D(__isnan) (x) != 0 && FUNC_D(__isnan) (y)) return DEC_NAN; //returning NaN
+  //If just one argument is NaN return the other argument
+  if( FUNC_D(__isnan) (x) != 0) return y; 
+  if( FUNC_D(__isnan) (y) != 0) return x;
+ 
+  //Line below works for normal numbers and +/-Infinity 
+  if( FUNC_D(__fabs) (x) == FUNC_D(__fabs) (y))
+  {
+    if (x > 0) return x;
+    return y;
+  }
+  if( FUNC_D(__isinf) (x) != 0) return x; 
+  if( FUNC_D(__isinf) (y) != 0) return y; 
```

Comment:
This should be implicit in the comparison below.

Reply:
Done!
https://github.com/libdfp/libdfp/pull/97/commits/94f9fa0e77ee0eabbf74c18c230f8e959f8f9c27